plugins/kotlin/jvm-debugger/core/src/org/jetbrains/kotlin/idea/debugger/core/stepping/KotlinRequestHint.kt

```diff
@@ -187,11 +196,55 @@ class KotlinStepIntoRequestHint(
                 lastWasKotlinFakeLineNumber = false
                 return STOP
             }
+
+            // Step over `component*` function calls to let destructured parameters be initialized
+            if (location != null && shouldStepOverParameterDestructuring(context, frameProxy, location)) {
+                // Switching to minimal stepping size allows us to not skip locations of interest
+                // in case of one-line lambdas.
+                currentSteppingSize = StepRequest.STEP_MIN
+                return StepRequest.STEP_OVER
+            }
+
             return super.getNextStepDepth(context)
         } catch (ignored: VMDisconnectedException) {
         } catch (e: EvaluateException) {
             LOG.error(e)
         }
         return STOP
     }
+
+    override fun getSize(): Int = currentSteppingSize
+}
+
+private fun shouldStepOverParameterDestructuring(context: SuspendContextImpl, frameProxy: StackFrameProxyImpl, location: Location): Boolean {
+    val method = location.safeMethod() ?: return false
+    if (method.isBridge) {
+        return false
+    }
+
+    val sourcePosition = context.debugProcess.positionManager.getSourcePosition(location) ?: return false
+    val destructuredParametersNames = runReadAction { collectDestructuredParametersNames(sourcePosition) }
+    if (destructuredParametersNames.isEmpty()) {
+        return false
+    }
+
+    val visibleVariablesNames = InlineStackTraceCalculator.calculateVisibleVariables(frameProxy)
+        .mapNotNull { it.name() }
+        .toSet()
+    return destructuredParametersNames.any { it !in visibleVariablesNames }
+}
+
+private fun collectDestructuredParametersNames(sourcePosition: SourcePosition): List<String> {
+    val lambda = sourcePosition.elementAt?.parentOfType<KtFunctionLiteral>() ?: return emptyList()
```

Comment:
Unlimited `parentOfType<KtFunctionLiteral>()` is dangerous. In the following case, stepping into the `run()` function broke.

```
import destructurableClasses.B

fun foo(f: (B) -> Unit) {
    val b = B()
    f(b)
}

fun main() {
    foo { (x, _, y, _) ->
        val task = object : Runnable {
            override fun run() {
                print("foo")
            }
        }
        // STEP_INTO: 1
        //Breakpoint!
        task.run()
    }
}
```